File: portal/src/util/label.test.ts
import { describe, it, expect } from "@jest/globals";
import { generateLabel } from "./label";

describe("generateLabel", () => {
  it("generates label", () => {
    expect(generateLabel("a")).toEqual("A");
    expect(generateLabel("a_pen")).toEqual("A Pen");
    expect(generateLabel("foobar")).toEqual("Foobar");
    expect(generateLabel("a_to_b")).toEqual("A to B");
    expect(generateLabel("a_b_c_d")).toEqual("A B C D");
    expect(generateLabel("a_b_🙂_d")).toEqual("A B 🙂 D");
  });
});

File: pkg/util/labelutil/label_test.go
package labelutil

import (
	"testing"

	. "github.com/smartystreets/goconvey/convey"
)

func TestLabel(t *testing.T) {
	Convey("Label", t, func() {
		So(Label("a"), ShouldEqual, "A")
		So(Label("a_pen"), ShouldEqual, "A Pen")
		So(Label("foobar"), ShouldEqual, "Foobar")
		So(Label("a_to_b"), ShouldEqual, "A to B")
		So(Label("a_b_c_d"), ShouldEqual, "A B C D")
	})
}
